R/doeResponseSurfaceMethodology.R

```diff
@@ -594,23 +588,24 @@ doeResponseSurfaceMethodology <- function(jaspResults, dataset, options, ...){
 
 
 
-.responseSurfaceTableCall <- function(jaspResults, options, rsm, dataset, position) {
+.responseSurfaceTableCall <- function(jaspResults, options, rsm, i, dataset, position) {
 
   rsm <- summary(rsm)
 
-  .responseSurfaceTable(jaspResults, options, dataset, rsm)
+  .responseSurfaceTable(jaspResults, options, rsm, i, dataset)
 
   return()
 }
 
-.responseSurfaceTable <- function(jaspResults, options, dataset, rsm) {
+.responseSurfaceTable <- function(jaspResults, options, rsm, i, dataset) {
 
   TableContainer <- createJaspContainer()
-  jaspResults[["TableContainer"]] <- TableContainer
+  jaspResults[[paste0("TableContainer", i)]] <- TableContainer
 
-  if (is.null(jaspResults[["TableContainer"]][["coef"]]))
-    CoefTable <- createJaspTable(gettext("RSM Coefficients"))
-  if (is.null(jaspResults[["TableContainer"]][["RSQTable"]]))
+  if (is.null(jaspResults[[paste0("TableContainer", i)]][["coef"]]))
+    CoefTable <- createJaspTable(gettext(paste0("RSM Coefficients for ",
+                                               options[["rsmResponseVariables"]][[i]])))
+  if (is.null(jaspResults[[paste0("TableContainer", i)]][["RSQTable"]]))
```

Comment:
so this won't reuse the state.

1. you're always creating a new jaspContainer called TableContainer
2. this new container will never contain CoefTable or RSQTable, since it's by default completely empty.

Instead, you should, for both the containers and tables, do

```r
if (is.null(jaspResults[[paste0("TableContainer", i)]])) {
  TableContainer <- createJaspContainer()
  jaspResults[[paste0("TableContainer", i)]] <- TableContainer
} else {
  TableContainer <- jaspResults[[paste0("TableContainer", i)]]
}
```
Note that the `else` step is very important here. In the case of 

```r
  if (is.null(jaspResults[[paste0("TableContainer", i)]][["coef"]]))
    CoefTable <- createJaspTable(gettext(paste0("RSM Coefficients for ",
                                               options[["rsmResponseVariables"]][[i]])))
```
what happens when `is.null(jaspResults[[paste0("TableContainer", i)]][["coef"]])` is FALSE? Then the table can be retrieved from the state, but you never create an object called `CoefTable` (nor `RSQTable`) and the remained of the code will crash.